### ISSUE TITLE
istio mTLS with STRICT mode does not work as expected for TCP zookeeper service.

### ISSUE THREAD

**Issue by kaushiksrinivas** [istio/istio#21095]
Trying to install zookeeper with istio side car and mTLS global policy of STRICT mode.
Environment : K8S
global mTLS : true 
mTLS mode : STRICT.

But we see below issues in the setup.
1. Issue with service entry:
    When installed with 3 zookeeper pods, we expect 3 serviceEntries to be created per fqdn of the 3 separate pods. for ex : zk-kfkafka-0.zk-kfkafka-headless.test.svc.cluster.local, zk-kfkafka-1.zk-kfkafka-headless.test.svc.cluster.local, zk-kfkafka-2.zk-kfkafka-headless.test.svc.cluster.local.
  But we see creating a single serviceEntry with hosts as "anyrandomstring.local" allows us to access all   
   the three zookeeper pods from the clients.
2. Issue with destination rules:
    When we create a SE per fqdn and DR per fqdn, we see upon describe of the zookeeper pod via istioctl command line below output which says no Destination Rules are applied for the client port 2181.
  
output of istioctl describe command:
    
istioctl x describe pod zk-kfkafka-0 -n test
Pod: zk-kfkafka-0
   Pod Ports: 2181 (ckaf-zookeeper-server), 2888 (ckaf-zookeeper-server), 3888 (ckaf-zookeeper-server), 15090 (istio-proxy)
Suggestion: add 'version' label to pod for Istio telemetry.
--------------------
Service: zk-kfkafka
   Port: client 2181/UnsupportedProtocol
   2181 is named "client" which does not follow Istio conventions
WARNING Pilot predicts TLS Conflict on zk-kfkafka-0 port 2181 (pod enforces mTLS, clients speak HTTP)
  There is no DestinationRule.  Check AuthenticationPolicy default/test
--------------------
Service: zk-kfkafka-headless
   Port: server 2888/UnsupportedProtocol
   2888 is named "server" which does not follow Istio conventions
   Port: leader-election 3888/UnsupportedProtocol
   3888 is named "leader-election" which does not follow Istio conventions
2888 WARNING Pilot predicts TLS Conflict on zk-kfkafka-0 port 2888 (pod enforces mTLS, clients speak HTTP)
  There is no DestinationRule.  Check AuthenticationPolicy default/test
3888 WARNING Pilot predicts TLS Conflict on zk-kfkafka-0 port 3888 (pod enforces mTLS, clients speak HTTP)
  There is no DestinationRule.  Check AuthenticationPolicy default/test

But we are able to access the zk pods only from the client machines having the side car injected, any other pods without side car is being rejected with certificate issues. 
  But when global destination rule of "*.local" which comes when global mTLS is enabled in istio-system namespace is used instead of creating specific DR, istioctl describe command shows below output with DRs applied as well.
    
istioctl x describe pod zk-kfkafka-0 -n test
Pod: zk-kfkafka-0
   Pod Ports: 2181 (ckaf-zookeeper-server), 2888 (ckaf-zookeeper-server), 3888 (ckaf-zookeeper-server), 15090 (istio-proxy)
Suggestion: add 'version' label to pod for Istio telemetry.
--------------------
Service: zk-kfkafka
   Port: client 2181/UnsupportedProtocol
   2181 is named "client" which does not follow Istio conventions
DestinationRule: zk-zktest-1global.istio-system for "*.local"
   Traffic Policy TLS Mode: ISTIO_MUTUAL
Pilot reports that pod is STRICT (enforces mTLS) and clients speak mTLS
--------------------
Service: zk-kfkafka-headless
   Port: server 2888/UnsupportedProtocol
   2888 is named "server" which does not follow Istio conventions
   Port: leader-election 3888/UnsupportedProtocol
   3888 is named "leader-election" which does not follow Istio conventions
2888 DestinationRule: zk-zktest-1global.istio-system for "*.local"
   Traffic Policy TLS Mode: ISTIO_MUTUAL
2888 Pilot reports that pod is STRICT (enforces mTLS) and clients speak mTLS
3888 DestinationRule: zk-zktest-1global.istio-system for "*.local"
   Traffic Policy TLS Mode: ISTIO_MUTUAL
3888 Pilot reports that pod is STRICT (enforces mTLS) and clients speak mTLS

zookeeper service is a TCP traffic.
Is TCP traffic with istio mTLS STRICT mode supported and honors all the DRs and SEs ?

**Comment by esnible** [comment]
Related: https://github.com/istio/istio/issues/19280

**Comment by aknakshay** [comment]
@esnible I and Kaushik are working on this problem. 
The issue that you have tagged is what we faced earlier to bring up zk. We have made the necessary changes and the quorum is up with mTLS.
The problem that he has highlighted is regarding the different combinations of service entry and destination rule, and how the operator is not marking those ports as mTLS STRICT ports whereas our preliminary checks show that it's communicating using mTLS.
